### DISCUSSION TITLE
DataGrid row styling based on a bound item's property value

### DISCUSSION THREAD

**Discussion by alaingaudreau** [AvaloniaUI/Avalonia.Xaml.Behaviors#107] [Q&A] [answered]
Hi,

What would be the best way to set a DataGrid row's styling if one of it's bound item's property is of a certain value.

To be more precice, I have a DataGrid bound to a AvaloniaList of Account objects which has a property named AccountType.

I'd like rows that have AccountType = 5 to be bold or perhaps insert a string infront of one of the cell's text for them to stand out from the regular accounttypes.

I've extended DataTriggerBehavior and set breakpoints to track what is getting passed to OnValueChanged() and can't figure out the proper <i:Interaction.Behaviors> structure in the xaml in order to evaluate Account.AccountType in the OnValueChanged() when binding the items to the datagrid.

Could someone with more AvaloniaBehaviors experience give me a hand please?

**Comment by maxkatz6** [comment, marked as answer]
It's not AvaloniaBehaviors related answer, but you could subscribe on DataGrid.LoadingRow event and set row's property or a style class depending on a data context.

**Reply by alaingaudreau** [reply]
Thanks @maxkatz6 , that actually works quite well and most likely a better way to go about it.

I was definitively overthinking the problem when there was such a simple and integrated way to do it built into Avalonia's DataGrid events.

For anyone who may face this situation and stumble on this thread...

You'd subscribe to your DataGrid's LoadingRow:

```
var datagrid = this.Get<DataGrid>("ItemDataGrid");
datagrid.LoadingRow += DataGrid_LoadingRow;
```

and to process the event

```
private void DataGrid_LoadingRow(object sender, DataGridRowEventArgs e)
    {
        if (sender is DataGrid)
        {
            var item = e.Row?.DataContext;
            if (item is Account)
            {
                Account current = (Account)item;
                
                if (current.AccountType == Account.TYPE_SUBGROUP)
                {
                    e.Row.Background = Brushes.Chartreuse;
                }
            }
            
        }
    }
```
